Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,86 @@ namespace TestImpact
             return testSuites;
         }
     } // namespace GTest
+
+    namespace Python
+    {
+        AZStd::vector<PythonTestEnumerationSuite> TestEnumerationSuitesFactory(
+            const AZStd::string& testEnumerationData)
+        {
+            size_t startLine = 0;
+            size_t endLine = 0;
+
+            AZStd::vector<PythonTestEnumerationSuite> pairs;
+            AZStd::map<AZStd::string, AZStd::vector<TestEnumerationSuite>> testSuiteMap;
+            AZStd::string previousTestFixture;
+
+            while ((startLine = testEnumerationData.find_first_not_of('\n', endLine)) != AZStd::string::npos)
+            {
+                enum MatchEntries
+                {
+                    FullMatch,
+                    ScriptPath,
+                    TestFixture,
+                    TestName,
+                    NumMatches
+                };
+
+                endLine = testEnumerationData.find('\n', startLine);
+                AZStd::string curLine = testEnumerationData.substr(startLine, endLine - startLine);
+
+                // This regex matches pytest test names, in the form ScriptPath::TestFixture::TestName. It will accept both parameterized
+                // and unparamaterized tests (with or without [] at the end).
+                const AZStd::basic_regex testNamePattern = AZStd::basic_regex("(.*py)::([A-Za-z_/\\-0-9]*)::(.*)");
+
+                // Steps through our string line by line. Assumes lines are split by newline characters.
+                AZStd::smatch matchResults;
```

Comment:
Suggestion: i think you might be able to move this inside the `if`, i.e.: `if(AZStd::smatch matchResults; AZStd::regex_search(curLine, matchResults, testNamePattern))`